Helm with Ingress included (#2173)
mendersoftware/mender-docs: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md

helm repo update
```

You can now install the Mender Server. You also have to decide how to expose
the Mender API Gateway service. This Helm Chart includes a configurable Ingress resource

> for your convenience. Some sample Ingress values are detailed on the later section.

Reviewer: ```suggestion
for your convenience. Some sample Ingress values are detailed in the later section.
```

[Load Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
to expose the Mender Server outside the Kubernetes cluster.

**Required configuration**:
* Redirection from HTTP to HTTPS on the Ingress is required
* If the health check is required, the internal path is `/ui`
* Either a valid certificate for the `${MENDER_SERVER_DOMAIN}` must exists as a secret named `mender-ingress-tls`, or
  you already have configured the [Cert-manager](../01.Kubernetes/docs.md#installation-of-cert-manager-optional) resource

> * The Mender helm chart is deployed with `api_gateway.env.SSL=false` option:

Reviewer: ```suggestion
* The Mender Helm chart is deployed with the `api_gateway.env.SSL=false` option:
```

  ```
  api_gateway:
    env:
      SSL: false
  ```

<!--AUTOVERSION: "Starting from the Mender Helm Chart version %"/ignore -->

> You can refer to your Infrastructure Provider's documentation for how to create an Ingress resource.

Reviewer: ```suggestion
You can refer to your Infrastructure Provider's documentation for creating an Ingress resource.
```

Starting from the Mender Helm Chart version 5.1.0, a sample Ingress resource is included.

For example, here's an Ingress for the AWS EKS Provider:

> <!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/ignore -->

Reviewer: ```suggestion
<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
```

```bash

> cat <<-EOF >> mender-3.6.0-build3.yml

Reviewer: ```suggestion
cat <<-EOF >> mender-3.6.1.yml
```

ingress:
  enabled: true
  annotations:
    cert-manager.io/issuer: "letsencrypt"
    alb.ingress.kubernetes.io/healthcheck-path: /ui/
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig":{ "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
    alb.ingress.kubernetes.io/backend-protocol: HTTP
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
    alb.ingress.kubernetes.io/load-balancer-attributes: routing.http2.enabled=true,idle_timeout.timeout_seconds=600
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/ssl-policy: ELBSecurityPolicy-TLS-1-2-2017-01
    alb.ingress.kubernetes.io/target-type: ip
  path: /
  extraPaths:
    - path: /
      backend:
        serviceName: ssl-redirect
        servicePort: use-annotation

  hosts:
    - ${MENDER_SERVER_DOMAIN}
  tls:
  # this secret must exists or it can be created from a working cert-manager instance
   - secretName: mender-ingress-tls
     hosts:
       - ${MENDER_SERVER_DOMAIN}
EOF
```

For example, here's an Ingress for Azure AKS Provider:

> <!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/ignore -->

Reviewer: ```suggestion
<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
```

Author: This is an Azure example, won't work on the EKS cluster for the Integration tests.

```bash

> cat <<-EOF >> mender-3.6.0-build3.yml

Reviewer: ```suggestion
cat <<-EOF >> mender-3.6.1.yml
```

ingress:
  enabled: true
  annotations:
    appgw.ingress.kubernetes.io/backend-protocol: http
    appgw.ingress.kubernetes.io/health-probe-path: /ui/
    appgw.ingress.kubernetes.io/request-timeout: "600"
    appgw.ingress.kubernetes.io/ssl-redirect: "true"
    kubernetes.io/ingress.class: azure/application-gateway
  path: /
  hosts:
    - ${MENDER_SERVER_DOMAIN}
  tls:
  # this secret must exists or it can be created from a working cert-manager instance
   - secretName: mender-ingress-tls
     hosts:
       - ${MENDER_SERVER_DOMAIN}
EOF
```

You can now update the Helm Chart with the included ingress:

> <!--AUTOVERSION: "helm upgrade --install mender mender/mender -f mender-%.yml"/ignore -->

Reviewer: ```suggestion
<!--AUTOVERSION: "helm upgrade --install mender mender/mender -f mender-%.yml"/integration -->
```

```bash

> helm upgrade --install mender mender/mender -f mender-3.6.0-build3.yml

Reviewer: ```suggestion
helm upgrade --install mender mender/mender -f mender-3.6.1.yml
```

[Load Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
to expose the Mender Server outside the Kubernetes cluster.

> **Required configuration**:

Reviewer 2: I am having doubts about the number of ** second level has two, right?

Author: Here it's bold: 

**Required configuration**:
* Redirection from HTTP to HTTPS on the Ingress is required
* If the health check is required, the internal path is `/ui`

* Redirection from HTTP to HTTPS on the Ingress is required

> * If the health check is required, the internal path is `/ui`

Reviewer 2: tons of passive voice, the test may stop you.

* Either a valid certificate for the `${MENDER_SERVER_DOMAIN}` must exists as a secret named `mender-ingress-tls`, or
  you already have configured the [Cert-manager](../01.Kubernetes/docs.md#installation-of-cert-manager-optional) resource
* The Mender Helm chart is deployed with the `api_gateway.env.SSL=false` option:
  ```
  api_gateway:
    env:
      SSL: false
  ```

<!--AUTOVERSION: "Starting from the Mender Helm Chart version %"/ignore -->
You can refer to your Infrastructure Provider's documentation for creating an Ingress resource.
Starting from the Mender Helm Chart version 5.1.0, a sample Ingress resource is included.

For example, here's an Ingress for the AWS EKS Provider:

<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
```bash

> cat <<-EOF >> mender-3.6.3.yml

Reviewer 2: this is `>>` (append) on purpose?
